### DISCUSSION TITLE
Access the underlying Vue and Quasar classes to pass in arguments at instantiation

### DISCUSSION THREAD

**Discussion by sergeyyurkov1** [zauberzeug/nicegui#1001] [Ideas / Feature Requests]
### Question

My point is that certain libraries such as Vue.Draggable can function standalone, as I was able to integrate it in a custom component by following the Nipple.js example without much trouble. However, if I were to use Pinia that relies heavily on internal parts of Vue, I would need to register it with Vue first, and as far as I can see, there is no way to do that with how NiceGUI abstracts away these JavaScript processes.

To illustrate my issue, here is how Pinia registers with Vue: https://github.com/vuejs/pinia/discussions/1051#discussioncomment-2166751
There is no way around it.

Thank you.

**Comment by falkoschindler** [comment]
Hi @sergeyyurkov1, this sounds interesting, also I don't quite understand the application, yet.
Can you describe why you would want to use Pinia (or a similar library) with NiceGUI?
And maybe you can give an example implementation assuming you _could_ register additional libraries with Vue?
